### PR TITLE
throw exceptions on parsing errors so that file that cause it can be …

### LINKED ISSUE #1323: history parser exceptions lack file info
Often I see great number of exceptions like this in indexer logs:

```
Jan 14, 2017 12:04:33 AM org.opensolaris.opengrok.history.SCCSHistoryParser initFields
WARNING: An error occured while parsing date
java.text.ParseException: Unparseable date: ":0/02/07 15:25:57"
        at java.text.DateFormat.parse(DateFormat.java:366)
        at org.opensolaris.opengrok.history.SCCSHistoryParser.initFields(SCCSHistoryParser.java:130)
        at org.opensolaris.opengrok.history.SCCSHistoryParser.getRevision(SCCSHistoryParser.java:149)
        at org.opensolaris.opengrok.history.SCCSHistoryParser.parseFile(SCCSHistoryParser.java:90)
        at org.opensolaris.opengrok.history.SCCSHistoryParser.parse(SCCSHistoryParser.java:68)
        at org.opensolaris.opengrok.history.SCCSRepository.getHistory(SCCSRepository.java:261)
        at org.opensolaris.opengrok.history.FileHistoryCache.get(FileHistoryCache.java:499)
```

The problem is that they are impossible to pair with actual file that caused it. Seems like bunch of history parsers exhibit the pattern of catching the `ParseException` and just logging it.

### PR DESCRIPTION
**Pull request by vladak**
…identified

fixes #1323

### CODE REVIEW

**Comment by vladak** [comment]
I was thinking to make the file information available in each history parser however:
  - most of the parsers can parse directories as well as files
  - the instantiation often looks like this:
    `return new SCCSHistoryParser().parse(file, this);`
    so there is no file info contained per se in the object

**Comment by vladak** [comment]
Need @tulinkry to look at this since I modifies his SVN date parsing test case.

**Comment by vladak** [comment]
The trouble with history parsers in general is that some of them were hiding problems and reporting them only to log (with WARNING). I think if there is a problem parsing history, HistoryException should be always raised and the caller should react accordingly.

That said, the passing of exceptions is a bit messy. There are some 3 kinds of history parsers:
  - simply read file from disk and interpret its contents (SCCS)
    - non-problematic, can manipulate/throw exceptions freely
  - execute external command and parse its output (Mercurial)
    - usually constrained by `processStream()` interface which can only throw IOException so if there is problem e.g. with parsing the date which normally results in ParseException, such has to be wrapped inside IOException
  - ditto plus interpret the output as XML (Subversion)
    - same set of problems as previous kind plus the XML parser can throw SAXException

File: test/org/opensolaris/opengrok/history/SubversionHistoryParserTest.java [hunk 1]
```diff
@@ -202,16 +204,13 @@ public void testDateFormats() {
                 HistoryEntry e = result.getHistoryEntries().get(0);
                 assertEquals(revId, e.getRevision());
                 assertEquals(author, e.getAuthor());
-
-                if (expectedException) {
-                    assertNull(e.getDate());
-                } else {
-                    assertEquals(new SimpleDateFormat(format).parse(date), e.getDate());
-                }
+                assertEquals(new SimpleDateFormat(format).parse(date), e.getDate());
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/history/SubversionHistoryParserTest.java]: This seems reasonable for me

File: test/org/opensolaris/opengrok/history/SubversionHistoryParserTest.java [hunk 2]
```diff
@@ -67,7 +67,9 @@ public void tearDown() {
      */
     @Test
     public void parseEmpty() throws Exception {
-        History result = instance.parse("");
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/history/SubversionHistoryParserTest.java]: Bad copyright

File: src/org/opensolaris/opengrok/history/GitRepository.java [hunk 3]
```diff
@@ -178,10 +178,10 @@ Executor getHistoryLogExecutor(final File file, String sinceRevision)
     }
 
     /**
-     * Formatter for rfc 2822 which allows (as rfc) the optional day at the
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/history/GitRepository.java]: Bad copyright

File: src/org/opensolaris/opengrok/history/AccuRevHistoryParser.java [hunk 4]
```diff
@@ -158,7 +158,11 @@ public void processStream(InputStream input) throws IOException {
                     date = df.parse(data[2]);
                     entry.setDate(date);
                 } catch (ParseException pe) {
-                    LOGGER.log(Level.WARNING, "Could not parse date: " + line, pe);
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/history/AccuRevHistoryParser.java]: Bad copyright

File: src/org/opensolaris/opengrok/history/ClearCaseHistoryParser.java [hunk 5]
```diff
@@ -78,8 +78,7 @@ public void processStream(InputStream input) throws IOException {
         String s;
         HistoryEntry entry = null;
         while ((s = in.readLine()) != null) {
-            if (!"create version".equals(s) &&
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/history/ClearCaseHistoryParser.java]: There is a missing copyright, one should be added

**Comment by vladak** [comment]
fixed the copyrights

**Comment by tulinkry** [comment]
I will modify the tests again anyway.
Seems ok to me now

**Pull request merged**